overlord/servicestate/quota_control.go

```diff
@@ -35,40 +33,21 @@ import (
 )
 
 var (
-	systemdVersion int
+	systemdAtLeast230 bool
 )
 
-// TODO: move to a systemd.AtLeast() ?
-func checkSystemdVersion() error {
-	vers, err := systemd.Version()
-	if err != nil {
-		return err
-	}
-	systemdVersion = vers
-	return nil
+func checkSystemdVersion() {
+	systemdAtLeast230 = systemd.AtLeast(230)
 }
 
 func init() {
-	if err := checkSystemdVersion(); err != nil {
-		logger.Noticef("failed to check systemd version: %v", err)
-	}
-}
-
-// MockSystemdVersion mocks the systemd version to the given version. This is
-// only available for unit tests and will panic when run in production.
-func MockSystemdVersion(vers int) (restore func()) {
-	osutil.MustBeTestBinary("cannot mock systemd version outside of tests")
-	old := systemdVersion
-	systemdVersion = vers
-	return func() {
-		systemdVersion = old
-	}
+	checkSystemdVersion()
 }
 
 func quotaGroupsAvailable(st *state.State) error {
 	// check if the systemd version is too old
-	if systemdVersion < 230 {
-		return fmt.Errorf("systemd version too old: snap quotas requires systemd 230 and newer (currently have %d)", systemdVersion)
```

Comment:
I wonder if not showing the detected version could bite us (e.g. if we suddenly encounter a version string we don't parse properly).

Reply:
Good point. I restored it, though I had to refactor things a bit.

Reply:
Thanks for the changes, I think they look good. Still a bit unsure about the placement of mocking per my other comment (but feel free to object ;)).

Reply:
Nope, I simply didn't get to it yet :-)